app/Middleware/Access.ts

```diff
@@ -0,0 +1,26 @@
+import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'
+import Role from 'App/Models/Role';
+import User from 'App/Models/User'
+
+export default class Access {
+  public async handle(
+    { auth, response }: HttpContextContract,
+    next: () => Promise<void>
+  ) {
+    let user = auth.user
+
+    await user?.load('roles');
+    const roles = user?.roles
+    let flag = true
+    for(let ele of roles) {
+      if(ele.title === 'User'){
+        flag = false;
+        await next()
+      }
+    }
```

Comment:
we can simplify this

```ts
const roles = user?.roles.map(role => role.title);
if (roles.includes(RoleEnum.CUSTOMER)) {
...
}
```

here, you might have noticed, i have used an enum

so for all roles, you can create enum and then use the enum instead of standard string value